demos/projects/PC/linux/config/demo_config.h

```diff
@@ -0,0 +1,188 @@
+// SPDX-License-Identifier: MIT
+
+#ifndef DEMO_CONFIG_H
+#define DEMO_CONFIG_H
+
+/* FreeRTOS config include. */
+#include "FreeRTOSConfig.h"
+
+/**************************************************/
+/******* DO NOT CHANGE the following order ********/
+/**************************************************/
+
+/* Include logging header files and define logging macros in the following order:
+ * 1. Include the header file "logging_levels.h".
+ * 2. Define the LIBRARY_LOG_NAME and LIBRARY_LOG_LEVEL macros depending on
+ * the logging configuration for DEMO.
+ * 3. Include the header file "logging_stack.h", if logging is enabled for DEMO.
+ */
+
+#include "logging_levels.h"
+
+/* Logging configuration for the Demo. */
+#ifndef LIBRARY_LOG_NAME
+    #define LIBRARY_LOG_NAME    "AzureIoTDemo"
+#endif
+
+#ifndef LIBRARY_LOG_LEVEL
+    #define LIBRARY_LOG_LEVEL    LOG_INFO
+#endif
+
+/* Prototype for the function used to print to console on Windows simulator
+ * of FreeRTOS.
+ * The function prints to the console before the network is connected;
+ * then a UDP port after the network has connected. */
+extern void vLoggingPrintf( const char * pcFormatString,
+                            ... );
+
+/* Map the SdkLog macro to the logging function to enable logging
+ * on Windows simulator. */
+#ifndef SdkLog
+    #define SdkLog( message )    vLoggingPrintf message
+#endif
+
+#include "logging_stack.h"
+
+/************ End of logging configuration ****************/
+
+/**
+ * @brief Enable Device Provisioning
+ * 
+ * @note To disable Device Provisioning undef this macro
+ *
+ */
+#define democonfigENABLE_DPS_SAMPLE
+
+#ifdef democonfigENABLE_DPS_SAMPLE
+
+/**
+ * @brief Provisioning service endpoint.
+ *
+ * @note https://docs.microsoft.com/en-us/azure/iot-dps/concepts-service#service-operations-endpoint
+ * 
+ */
+#define democonfigENDPOINT                  "global.azure-devices-provisioning.net"
+
+/**
+ * @brief Id scope of provisioning service.
+ * 
+ * @note https://docs.microsoft.com/en-us/azure/iot-dps/concepts-service#id-scope
+ * 
+ */
+#define democonfigID_SCOPE                  "<YOUR ID SCOPE HERE>"
+
+/**
+ * @brief Registration Id of provisioning service
+ * 
+ * @warning If using X509 authentication, this MUST match the Common Name of the cert.
+ *
+ *  @note https://docs.microsoft.com/en-us/azure/iot-dps/concepts-service#registration-id
+ */
+#define democonfigREGISTRATION_ID           "<YOUR REGISTRATION ID HERE>"
+
+#endif // democonfigENABLE_DPS_SAMPLE
+
+/**
+ * @brief IoTHub device Id.
+ *
+ */
+#define democonfigDEVICE_ID                 "<YOUR DEVICE ID HERE>"
+
+/**
+ * @brief IoTHub module Id.
+ *
+ */
+#define democonfigMODULE_ID                 ""
+/**
+ * @brief IoTHub hostname.
+ *
+ */
+#define democonfigHOSTNAME                  "<YOUR IOT HUB HOSTNAME HERE>"
+
+/**
+ * @brief Device symmetric key
+ *
+ */
+#define democonfigDEVICE_SYMMETRIC_KEY      "<Symmetric key>"
```

Comment:
Could we make it more clear - eventually, OK if it's not this PR but we should track work somehow - that it's either Sym key or else x509?  That doesn't necessarly jump out at me way that the code is currently commented.

Reply:
sample does have check that either Symmetric key or Cert is define.

https://github.com/Azure-Samples/iot-middleware-freertos-samples/blob/dae616d0c665684590f60252ad6d6442608d4531/demos/sample_azure_iot/sample_azure_iot.c#L46